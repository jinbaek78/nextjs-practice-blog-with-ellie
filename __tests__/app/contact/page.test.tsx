import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import ContactPage, { LINKS } from '@/app/contact/page';

describe.skip('Contact Page', () => {
  it('renders', () => {
    const { container } = render(<ContactPage />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<ContactPage />);
    expect(screen.getAllByRole('link')).toHaveLength(LINKS.length);
  });
});
