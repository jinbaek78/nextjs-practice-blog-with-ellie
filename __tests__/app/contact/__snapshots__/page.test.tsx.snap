// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page renders 1`] = `
<div>
  <section
    class="flex flex-col items-center"
  >
    <h2
      class="text-3xl font-bold my-2"
    >
      Contact Me
    </h2>
    <p>
      info@dream-coding.com
    </p>
    <ul
      class="flex gap-4 my-2"
    >
      <a
        class="text-5xl hover:text-yellow-400"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
          />
        </svg>
      </a>
      <a
        class="text-5xl hover:text-yellow-400"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
          />
        </svg>
      </a>
      <a
        class="text-5xl hover:text-yellow-400"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M941.3 296.1a112.3 112.3 0 0 0-79.2-79.3C792.2 198 512 198 512 198s-280.2 0-350.1 18.7A112.12 112.12 0 0 0 82.7 296C64 366 64 512 64 512s0 146 18.7 215.9c10.3 38.6 40.7 69 79.2 79.3C231.8 826 512 826 512 826s280.2 0 350.1-18.8c38.6-10.3 68.9-40.7 79.2-79.3C960 658 960 512 960 512s0-146-18.7-215.9zM423 646V378l232 133-232 135z"
          />
        </svg>
      </a>
    </ul>
    <h2>
      Or Send me an email
    </h2>
    <section
      class="w-full max-w-md "
    >
      <form
        class="w-full  flex flex-col gap-2 m-4 p-4 bg-slate-700 rounded-xl text-white"
      >
        <label
          class="font-semibold"
          for="from"
        >
          Your Email
        </label>
        <input
          class="text-black"
          data-testid="from"
          id="from"
          name="from"
          required=""
          type="email"
          value=""
        />
        <label
          class="font-semibold"
          for="subject"
        >
          Subject
        </label>
        <input
          class="text-black"
          data-testid="subject"
          id="subject"
          name="subject"
          required=""
          type="text"
          value=""
        />
        <label
          class="font-semibold"
          for="message"
        >
          Message
        </label>
        <textarea
          class="text-black"
          data-testid="message"
          id="message"
          name="message"
          required=""
          rows="10"
        />
        <button
          class="bg-yellow-300 text-black font-bold hover:bg-yellow-400"
          type="submit"
        >
          Submit
        </button>
      </form>
    </section>
  </section>
</div>
`;
